app/controllers/application_controller.rb

```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::API
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  private
+
+  def authenticate
+    authenticate_or_request_with_http_token do |token, options|
+      decoded_token = JWT.decode(token, ENV['SECRET_KEY'], true, algorithm: 'HS256')
```

Comment:
Acredito que não precisamos adicionar o retorno em uma variável já que não estamos a utilizando, o que você acha?
```suggestion
      JWT.decode(token, ENV['SECRET_KEY'], true, algorithm: 'HS256')
```